src/core/server/http/router/router.ts

```diff
@@ -138,6 +139,43 @@ function routeSchemasFromRouteConfig<
   return route.validate ? route.validate : undefined;
 }
 
+/**
+ * Create a valid options object with "sensible" defaults + adding some validation to the options fields
+ *
+ * @param method HTTP verb for these options
+ * @param routeConfig The route config definition
+ */
+function validOptions<Method extends RouteMethod>(
+  method: Method,
+  routeConfig: RouteConfig<ObjectType, ObjectType, ObjectType, Method>
+) {
+  const shouldNotHavePayload = ['head', 'get'].includes(method);
+  const { options = {}, validate } = routeConfig;
+  const shouldValidateBody = validate && validate.body;
+
+  const { output } = options.body || {};
+  if (typeof output === 'string' && !['data', 'stream'].includes(output)) {
```

Comment:
nit: easy to forget to update this logic. Shouldn't we extract in own type / enum? https://github.com/elastic/kibana/pull/50783/files#diff-3a561a752ff7c0106ea1ce21c8a9ea49R76

Reply:
Makes perfect sense! Let me change this